<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1157</width>
    <height>581</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(255, 255, 255);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>110</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QWidget" name="Input_widget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>0</y>
      <width>701</width>
      <height>361</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(219, 253, 218);</string>
    </property>
    <widget class="QPushButton" name="choose_data_btn">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Choose data</string>
     </property>
    </widget>
    <widget class="QTableView" name="X_table">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>271</width>
       <height>281</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="data_path">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>20</y>
       <width>241</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>100</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Tham số đầu vào</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>130</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Số phần tử giám sát</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>160</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Số cụm</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="k_text">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>130</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="c_text">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>160</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="epsilon_text">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>190</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>190</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Epsilon</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="m_text">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>250</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>260</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>m</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="M_text">
     <property name="geometry">
      <rect>
       <x>480</x>
       <y>250</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>250</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>M</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="M1_text">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>250</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>570</x>
       <y>250</y>
       <width>20</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>M'</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="algo1_check">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>220</y>
       <width>81</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>sSFCM</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="algo2_check">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>220</y>
       <width>81</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>sSMC-FCM</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_2pha_algo2">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>280</y>
       <width>70</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>2 Pha ?</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_2pha_algo1">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>280</y>
       <width>70</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>2 Pha ?</string>
     </property>
    </widget>
    <widget class="QPushButton" name="run_btn">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>310</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>RUN</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>50</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Cột chứa nhãn dữ liệu</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="label_column">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>70</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>640</x>
       <y>60</y>
       <width>31</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>OK</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>10</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Hiệu chỉnh dữ liệu</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>40</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Cột bắt đầu</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>60</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>70</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Hàng bắt đầu</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="begin_col">
     <property name="geometry">
      <rect>
       <x>540</x>
       <y>40</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="begin_row">
     <property name="geometry">
      <rect>
       <x>540</x>
       <y>70</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>380</y>
      <width>531</width>
      <height>191</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(219, 253, 218);</string>
    </property>
    <widget class="QTableWidget" name="validity_table">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>531</width>
       <height>121</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:rgb(255, 255, 255);</string>
     </property>
     <property name="showGrid">
      <bool>true</bool>
     </property>
     <property name="gridStyle">
      <enum>Qt::SolidLine</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <row>
      <property name="text">
       <string>Rand Index</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>Adjusted Rand Index</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>Jaccard Coefficient</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>sSFCM(1pha)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>sSFCM(2pha)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>sSMC-FCM(1pha)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>sSMC-FCM(2pha)</string>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string/>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>10</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font-size: 20px;</string>
     </property>
     <property name="text">
      <string>External_validity</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>380</y>
      <width>551</width>
      <height>181</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(219, 253, 218);</string>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>sSFCM</string>
     </attribute>
     <widget class="QTabWidget" name="tabWidget_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>571</width>
        <height>161</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:rgb(219, 253, 218);</string>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_7">
       <attribute name="title">
        <string>V(1 Pha)</string>
       </attribute>
       <widget class="QTableView" name="V_table_1">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>521</width>
          <height>111</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_8">
       <attribute name="title">
        <string>V(2 Pha)</string>
       </attribute>
       <widget class="QTableView" name="V_table_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>521</width>
          <height>111</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>sSMC-FCM</string>
     </attribute>
     <widget class="QTabWidget" name="tabWidget_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>551</width>
        <height>161</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_9">
       <attribute name="title">
        <string>V(1 Pha)</string>
       </attribute>
       <widget class="QTableView" name="V_table_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>521</width>
          <height>111</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_10">
       <attribute name="title">
        <string>V(2 Pha)</string>
       </attribute>
       <widget class="QTableView" name="V_table_4">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>521</width>
          <height>111</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget_4">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>10</y>
      <width>391</width>
      <height>341</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(219, 253, 218);</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>sSFCM</string>
     </attribute>
     <widget class="QTabWidget" name="tab_17">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>511</width>
        <height>321</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="tab_11">
       <attribute name="title">
        <string>U_ngang(1 Pha)</string>
       </attribute>
       <widget class="QTableView" name="U_ngang_table_1">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>281</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_15">
       <attribute name="title">
        <string>U(1 Pha)</string>
       </attribute>
       <widget class="QTableView" name="U_table_1">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>281</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_12">
       <attribute name="title">
        <string>U_ngang(2 Pha)</string>
       </attribute>
       <widget class="QTableView" name="U_ngang_table_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>281</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_16">
       <attribute name="title">
        <string>U(2 Pha)</string>
       </attribute>
       <widget class="QTableView" name="U_table_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>281</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>sSMC-FCM</string>
     </attribute>
     <widget class="QTabWidget" name="tab_18">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>511</width>
        <height>311</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>M(1 Pha)</string>
       </attribute>
       <widget class="QTableView" name="M_table_1">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>281</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_13">
       <attribute name="title">
        <string>U(1 Pha)</string>
       </attribute>
       <widget class="QTableView" name="U_table_3">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>301</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_6">
       <attribute name="title">
        <string>M(2 Pha)</string>
       </attribute>
       <widget class="QTableView" name="M_table_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>301</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_14">
       <attribute name="title">
        <string>U(2 Pha)</string>
       </attribute>
       <widget class="QTableView" name="U_table_4">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>371</width>
          <height>281</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color:rgb(255, 255, 255);</string>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
   </widget>
   <zorder>label</zorder>
   <zorder>widget</zorder>
   <zorder>tabWidget</zorder>
   <zorder>tabWidget_4</zorder>
   <zorder>Input_widget</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1157</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>choose_data_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>choose_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>run_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>caculate_cluster()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>407</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>preprocess_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>578</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>choose_data()</slot>
  <slot>caculate_cluster()</slot>
  <slot>preprocess_data()</slot>
 </slots>
</ui>
